pyte/streams.py

```diff
@@ -334,7 +334,9 @@ def create_dispatcher(mapping):
                 param = ""
                 while True:
                     char = yield
-                    if char == ST or char == ctrl.BEL:
+                    if char == ESC:
+                        char += yield
+                    if char in [ctrl.ST_C0, ctrl.ST_C1, ctrl.BEL]:
```

Comment:
Have you considered using chained `==`?

Reply:
Do you mean like this?

```python
if char == ctrl.ST_C0 or char == ctrl.ST_C1 or char == ctrl.BEL:
```

I opted against this as it seemed a little repetitive and would need to be wrapped to fit in 79 columns.

Maybe I could make a list or set called `OSC_TERMINATORS` or similar earlier in the function after `ALLOWED_IN_OSC` to fit more closely with the existing code. I could also change the existing groups of characters to be sets rather than strings, for consistency. I did a little timing and found the set lookups to be only marginally less performant than a lookup in a string with 6 characters.

Reply:
Thanks, I'd suggest going for a lengthy == version or `OSC_TERMINATORS`. I can do it after merging, if that's okay with you.